### PR TITLE
Implement a basic recursive_delete_if that passes the failing test.

### PR DESCRIPTION
**Pull request by parndt**
WDYT?

### CODE REVIEW

**Comment by parndt** [comment]
The naming of the method, the placement of the method, all these things could be improved. This is more of a proof of concept.

**Comment by steveklabnik** [comment]
Looks good to me. :heart:

File: lib/json/merge_patch.rb [hunk 1]
```diff
@@ -18,10 +18,19 @@ def self.merge(document, merge_patch)
...
+  rescue
     raise MergeError
   end
+
+  def self.recursive_delete_if(key, value)
+    if value.kind_of?(Hash)
+      value.delete_if(&method(:recursive_delete_if))
+      nil
+    else
+      value.nil?
```
Comment (parndt) [inline lib/json/merge_patch.rb:33]: I wonder about using `value.empty?` here.

Comment (steveklabnik) [inline lib/json/merge_patch.rb:33]: The spec says a `null` in JSON. An empty string is different.

Comment (parndt) [inline lib/json/merge_patch.rb:33]: Ah, sweet as.

Comment (steveklabnik) [inline lib/json/merge_patch.rb:33]: Yup.

Now, the final pending test isn't passing, even though it should be. :cry: so close!

**Pull request merged**